### PR TITLE
MCR-3210 improve transaction system

### PR DESCRIPTION
**Pull request by Mewel**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3210).

### MCRTransactionManager
**Centralized Transaction Management:** The MCRTransactionManager ensures that all transaction-related operations—beginning, committing, and rolling back—are managed centrally. This eliminates scattered transaction management logic and consolidates error handling, which improves the overall reliability of the system.

**Error Handling and Rollback Mechanism:** If any transaction fails during the commit or rollback process, MCRTransactionManager attempts to roll back any previously started transactions and suppresses any rollback exceptions. This helps prevent partial commits and leaves the system in a consistent state.

**Transaction Prioritization:** The manager allows setting a commit priority for each transaction, ensuring that more critical transactions are committed before others, thus enabling finer control over the transaction lifecycle.

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/common/MCRTransactionManager.java [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+     * @return the currently active {@link TransactionLoader}.
+     */
+    public static TransactionLoader getTransactionLoader() {
+        return transactionLoader;
+    }
+
+    /**
+     * Begins all available {@link MCRPersistenceTransaction} instances that are active.
+     *
+     * <p>It throws an exception if there are already active transactions in the current thread.</p>
```
Comment (sebhofmann) [inline mycore-base/src/main/java/org/mycore/common/MCRTransactionManager.java]: This is already stated in the `@throws ` part.

**Review by sebhofmann — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/common/MCRSession.java [hunk 2]
```diff
@@ -407,7 +407,7 @@ public long getCreateTime() {
      */
     @Deprecated
```
Comment (rsteph-de) [inline mycore-base/src/main/java/org/mycore/common/MCRSession.java:408]: Deprecated in 2021.06 - time to delete?

Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/common/MCRSession.java:408]: Not in this PR. Let's do this together and remove deprecated methods all together in main AFTER we forked release branch

File: mycore-base/src/main/java/org/mycore/backend/jpa/MCREntityTransaction.java [hunk 3]
```diff
@@ -45,17 +45,8 @@ public void rollback() {
     }
 
     @Override
-    public boolean getRollbackOnly() {
-        return MCREntityManagerProvider.getCurrentEntityManager().getTransaction().getRollbackOnly();
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/backend/jpa/MCREntityTransaction.java:49]: A library may support transaction that we want to delegate. If an error occurs the transaction of the library may signal via getRollbackOnly() that a commit will fail anyway. So why do we drop getRollbackOnly()?

Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/backend/jpa/MCREntityTransaction.java:49]: It's likely that this is only needed, if JPA transaction handling is done via other libraries/container. So we could test the code as suggested.

**Review by yagee-de — APPROVED** [review]

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**